File: GoHttp-uTLS/ClientHello.cs
namespace GoHttp_uTLS
{
    public enum ClientHello
    {
        // HelloGolang
        HelloGolang = 10_001_00,

        // HelloCustom
        HelloCustom = 11_001_00,

        // HelloRandomized
        HelloRandomized = 12_001_00,
        HelloRandomizedALPN = 12_002_00,
        HelloRandomizedNoALPN = 12_003_00,

        // HelloFirefox
        HelloFirefox_55 = 13_055_00,
        HelloFirefox_56 = 13_056_00,
        HelloFirefox_63 = 13_063_00,
        HelloFirefox_65 = 13_065_00,
        HelloFirefox_99 = 13_099_00,
        HelloFirefox_102 = 13_102_00,
        HelloFirefox_105 = 13_105_00,
        HelloFirefox_Auto = HelloFirefox_105,

        // HelloChrome
        HelloChrome_58 = 14_058_00,
        HelloChrome_62 = 14_062_00,
        HelloChrome_70 = 14_070_00,
        HelloChrome_72 = 14_072_00,
        HelloChrome_83 = 14_083_00,
        HelloChrome_87 = 14_087_00,
        HelloChrome_96 = 14_096_00,
        HelloChrome_100 = 14_100_00,
        HelloChrome_102 = 14_102_00,
        HelloChrome_Auto = HelloChrome_102,

        // HelloIOS
        HelloIOS_11_1 = 15_011_01,
        HelloIOS_12_1 = 15_012_01,
        HelloIOS_13 = 15_013_00,
        HelloIOS_14 = 15_014_00,
        HelloIOS_Auto = HelloIOS_14,

        // HelloAndroid
        HelloAndroid_11_OkHttp = 16_011_00,

        // HelloEdge
        HelloEdge_85 = 17_085_00,
        HelloEdge_106 = 17_106_00,
        HelloEdge_Auto = HelloEdge_85,

        // HelloSafari
        HelloSafari_16_0 = 18_016_00,
        HelloSafari_Auto = HelloSafari_16_0,

        // Hello360
        Hello360_7_5 = 19_007_05,
        Hello360_11_0 = 19_011_00,
        Hello360_Auto = Hello360_7_5,

        // HelloQQ
        HelloQQ_11_1 = 20_011_01,
        HelloQQ_Auto = HelloQQ_11_1
    }
}

File: GoHttpLib/ClientHello.go
package main

import (
	tls "github.com/refraction-networking/utls"
)

func GetClientHelloID(id int) *tls.ClientHelloID {
	// tls.HelloGolang
	if id == 10_001_00 {
		return &tls.HelloGolang
	}

	// tls.HelloCustom
	if id == 11_001_00 {
		return &tls.HelloCustom
	}

	// tls.HelloRandomized
	if id == 12_001_00 {
		return &tls.HelloRandomized
	}
	if id == 12_002_00 {
		return &tls.HelloRandomizedALPN
	}
	if id == 12_003_00 {
		return &tls.HelloRandomizedNoALPN
	}

	// tls.HelloFirefox
	if id == 13_055_00 {
		return &tls.HelloFirefox_55
	}
	if id == 13_056_00 {
		return &tls.HelloFirefox_56
	}
	if id == 13_063_00 {
		return &tls.HelloFirefox_63
	}
	if id == 13_065_00 {
		return &tls.HelloFirefox_65
	}
	if id == 13_099_00 {
		return &tls.HelloFirefox_99
	}
	if id == 13_102_00 {
		return &tls.HelloFirefox_102
	}
	if id == 13_105_00 {
		return &tls.HelloFirefox_105
	}

	// tls.HelloChrome
	if id == 14_058_00 {
		return &tls.HelloChrome_58
	}
	if id == 14_062_00 {
		return &tls.HelloChrome_62
	}
	if id == 14_070_00 {
		return &tls.HelloChrome_70
	}
	if id == 14_072_00 {
		return &tls.HelloChrome_72
	}
	if id == 14_083_00 {
		return &tls.HelloChrome_83
	}
	if id == 14_087_00 {
		return &tls.HelloChrome_87
	}
	if id == 14_096_00 {
		return &tls.HelloChrome_96
	}
	if id == 14_100_00 {
		return &tls.HelloChrome_100
	}
	if id == 14_102_00 {
		return &tls.HelloChrome_102
	}

	// tls.HelloIOS
	if id == 15_011_01 {
		return &tls.HelloIOS_11_1
	}
	if id == 15_012_01 {
		return &tls.HelloIOS_12_1
	}
	if id == 15_013_00 {
		return &tls.HelloIOS_13
	}
	if id == 15_014_00 {
		return &tls.HelloIOS_14
	}

	// tls.HelloAndroid
	if id == 16_011_00 {
		return &tls.HelloAndroid_11_OkHttp
	}

	// tls.HelloEdge
	if id == 17_085_00 {
		return &tls.HelloEdge_85
	}
	if id == 17_106_00 {
		return &tls.HelloEdge_106
	}

	// tls.HelloSafari
	if id == 18_016_00 {
		return &tls.HelloSafari_16_0
	}

	// tls.Hello360
	if id == 19_007_05 {
		return &tls.Hello360_7_5
	}
	if id == 19_011_00 {
		return &tls.Hello360_11_0
	}

	// tls.HelloQQ
	if id == 20_011_01 {
		return &tls.HelloQQ_11_1
	}
	return nil
}
